core/sql/optimizer/NATable.cpp

```diff
@@ -275,75 +282,28 @@ void HistogramCache::getHistograms(NATable& table)
     // look up the cache and get a reference to statistics for this table
     cachedHistograms = lookUp(table);
 
-    // first thing to check is, if the table to which the histograms are cached
-    // has been updated
+    // (Possibly useless) sanity tests
+
+    // Check to see if the redefinition timestamp has changed. This seems
+    // to be always stubbed to zero today on Trafodion, so this check
+    // seems to never fail.
 
     if (cachedHistograms && (cachedHistograms->getRedefTime() != redefTime))
     {
       deCache(&cachedHistograms);
     }
```

Comment:
I actually changed DDL code to use the non-zero redeftime in descriptors as part of my role invalidation change, but it caused some interesting regression failures.  So I removed the change before delivering the code; hopefully, I can get back to it sometime.